src/main/java/lberkholtz/advancedjava/assignmentseven/util/DatabaseUtils.java

```diff
@@ -0,0 +1,267 @@
+            // an example of throwing an exception appropriate to the abstraction.
+        } catch (ClassNotFoundException | SQLException e) {
+            String message = e.getMessage();
+            throw new DatabaseConnectionException("Could not connection to model." + message, e);
+        }
+        return connection;
+    }
+
+    /**
+     * A utility method that runs a db initialize script.
+     *
+     * @param initializationScript full path to the script to run to create the schema
+     * @throws DatabaseInitializationException
+     */
+    public static void initializeDatabase(String initializationScript) throws DatabaseInitializationException {
+
+        Connection connection = null;
+        final StringBuilder errorLog = new StringBuilder();
+        try {
+            connection = getConnection();
+            connection.setAutoCommit(false);
+            ScriptRunner runner = new ScriptRunner(connection, false, false);
+            InputStream inputStream = new FileInputStream(initializationScript);
+
+            InputStreamReader reader = new InputStreamReader(inputStream);
+
+            runner.setErrorLogWriter(new PrintWriter(new OutputStream() {
+                @Override
+                public void write(int b) throws IOException {
+                    errorLog.append((char) b);
+                }
+            }));
+            runner.runScript(reader);
+
+            reader.close();
+            connection.commit();
+            connection.close();
+
+        } catch (DatabaseConnectionException | SQLException | IOException e) {
+            throw new DatabaseInitializationException("Could not initialize db because of:"
+                    + e.getMessage(), e);
+        }
+        // improvement - previous versions would not single errors in the SQL itself.
+        if (errorLog.length() != 0) {
+            throw new DatabaseInitializationException("SQL init script contained errors:" + errorLog.toString());
+
+
+        }
+    }
+
+    /**
+     * Execute SQL code
+     *
+     * @param someSQL the code to execute
+     * @return true if the operation succeeded.
+     * @throws DatabaseException if accessing and executing the sql failed in an unexpected way.
+     */
+    public static boolean executeSQL(String someSQL) throws DatabaseException {
+        Connection connection = null;
+        boolean returnValue = false;
+        try {
+            connection = DatabaseUtils.getConnection();
+            Statement statement = connection.createStatement();
+            returnValue = statement.execute(someSQL);
+        } catch (DatabaseConnectionException | SQLException e) {
+            throw new DatabaseException(e.getMessage(), e);
+        }
+        return returnValue;
+    }
+
+
+    /*
+   * @return SessionFactory for use with model transactions
+   */
+    public static SessionFactory getSessionFactory() {
+
+        // singleton pattern
+        synchronized (DatabaseUtils.class) {
```

Comment:
normally synchronize of SessionFactory (the same class as where the synchronized is used)